src/shared/parameter_v2/examples/generated_dynamic_parameters.hpp

```diff
@@ -0,0 +1,395 @@
+        
+        boost::program_options::variables_map vm;
+        boost::program_options::store(parse_command_line(argc, argv, desc), vm);
+        boost::program_options::notify(vm);
+
+        this->mutableFooBool()->setValue(args.foo_bool);
+        this->mutableFooInt()->setValue(args.foo_int);
+
+        if (args.help)
+        {
+            std::cout << desc << std::endl;
+        }
+
+        return args.help;
+    }
+
+    const MutableParameterList& getMutableParameterList()
+    {
+        return mutable_internal_param_list;
+    }
+
+    const ParameterList& getParameterList() const
+    {
+        return immutable_internal_param_list;
+    }
+
+   private:
+    MutableParameterList mutable_internal_param_list;
+    ParameterList immutable_internal_param_list;
+    std::shared_ptr<Parameter<bool>> foo_bool;
+    std::shared_ptr<NumericParameter<int>> foo_int;
+}
+
+class exampleConfig : public Config
+{
+   public:
+    exampleConfig(std::shared_ptr<fooConfig> foo_config) : foo_config(foo_config)
+    {
+        example_bool_param = std::make_shared<Parameter<bool>>("example_bool_param", true);
+        example_int_param = std::make_shared<NumericParameter<int>>("example_int_param", 3, 0, 5);
+        example_unsigned_int_param = std::make_shared<NumericParameter<uint>>("example_unsigned_int_param", 3, 0, 5);
+        example_float_param = std::make_shared<NumericParameter<float>>("example_float_param", 4.04, 1.1, 9.01);
+        example_string_param = std::make_shared<Parameter<std::string>>("example_string_param", "Hello World");
+        example_enum_param = std::make_shared<EnumeratedParameter<RefereeCommand>>("example_enum_param", RefereeCommand.HALT, {RefereeCommand.HALT, RefereeCommand.START});
+        example_factory_param = std::make_shared<EnumeratedParameter<std::string>>("example_factory_param", "HaltPlay", GenericFactory<std::string, Play>::getRegisteredNames());
+        example_optional_param = std::make_shared<Parameter<std::optional<bool>>>("example_optional_param", false);
+        mutable_internal_param_list = {
+            example_bool_param,
+            example_int_param,
+            example_unsigned_int_param,
+            example_float_param,
+            example_string_param,
+            example_enum_param,
+            foo_config,
+        };
+        immutable_internal_param_list = {
+            std::const_pointer_cast<const Parameter<bool>>(example_bool_param),
+            std::const_pointer_cast<const NumericParameter<int>>(example_int_param),
+            std::const_pointer_cast<const NumericParameter<uint>>(example_unsigned_int_param),
+            std::const_pointer_cast<const NumericParameter<float>>(example_float_param),
+            std::const_pointer_cast<const Parameter<std::string>>(example_string_param),
+            std::const_pointer_cast<const EnumeratedParameter<RefereeCommand>>(example_enum_param),
```

Comment:
there's some discussion on slack about representing enums as strings for dynamic params. I think it's relevant here

Reply:
Made changes as discussed